File: Python/PE-5.py
'''2520 is the smallest number that can be divided
by each of the numbersfrom 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible
by all of the numbers from 1 to 20?
'''
from math import gcd


def main():
    n = 20

    lcm = 1

    for number in range(2, n + 1):
        lcm = (lcm * number) // gcd(lcm, number)
    
    print(lcm)


if __name__ == '__main__':
    main()

File: Go/PE-5/PE-5.go
package main

import (
	"fmt"
)

func gcd(a int, b int) int {
	for b != 0 {
		a, b = b, a%b
	}

	return a
}

func main() {
	n := 20

	lcm := 1

	for i := 2; i <= n; i++ {
		lcm = (lcm * i) / gcd(lcm, i)
	}

	fmt.Println(lcm)
}
